### PR TITLE
Fix signature of decorated subclass of abstract `Named` subclass

### LINKED ISSUE #27108: Cryptic error message when instantiating type that fails to implement interface methods
### Current Behavior

Consider the following minimal reproducer:
```
given:
buildFile << """
    abstract class Reproducer implements Named {
        @Inject
        Reproducer() {}
    }
    objects.newInstance(Reproducer)
"""

expect:
succeeds("help")
```

This fails with:
```
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.doCreate(DependencyInjectingInstantiator.java:69)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.newInstance(DependencyInjectingInstantiator.java:55)
	at org.gradle.api.internal.model.DefaultObjectFactory.newInstance(DefaultObjectFactory.java:90)
	at build_1slav2hx8m13lbvv1i8qw5ki6.run(gradle/platforms/software/dependency-management/build/tmp/teŝt files/Reproducer/fs6a6/build.gradle:6)
	at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:91)
	... 150 more
Caused by: java.lang.reflect.GenericSignatureFormatError: Signature Parse error: expected '('
	Remaining input: Ljava/lang/String;()V
	at org.gradle.internal.instantiation.generator.AbstractClassGenerator$GeneratedClassImpl$GeneratedConstructorImpl.getGenericParameterTypes(AbstractClassGenerator.java:554)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.addServicesToParameters(DependencyInjectingInstantiator.java:162)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.convertParameters(DependencyInjectingInstantiator.java:121)
	at org.gradle.internal.instantiation.generator.DependencyInjectingInstantiator.doCreate(DependencyInjectingInstantiator.java:62)
	... 154 more
```

### Expected Behavior

An error message that better guides the user to a solution

### Context (optional)

_No response_

### Steps to Reproduce

See above

### Gradle version

8.4

### Build scan URL (optional)

_No response_

### Your Environment (optional)

_No response_

### PR DESCRIPTION
**Pull request by bamboo**
Fixes #27108

### CODE REVIEW

File: platforms/core-configuration/model-core/src/main/java/org/gradle/model/internal/asm/AsmClassGeneratorUtils.java [hunk 1]
```diff
@@ -39,14 +40,20 @@ public static String signature(Constructor<?> constructor) {
      */
     public static String signature(Constructor<?> constructor, boolean addNameParameter) {
```
Comment (abstratt) [inline platforms/core-configuration/model-core/src/main/java/org/gradle/model/internal/asm/AsmClassGeneratorUtils.java:41]: Looks good!

This method and the variant that takes a Method strike me as something that would be better unit-tested than integration-tested (much easier to exercise different scenarios, easier to understand expectations too).

Also, it seems the other variant of this method (which takes only the constructor) is not used anywhere.

**Review by abstratt — APPROVED** [review]

**Comment by bamboo** [comment]
@bot-gradle test

File: platforms/core-configuration/model-core/src/main/java/org/gradle/model/internal/asm/AsmClassGeneratorUtils.java (hunk 1 above)
Comment (bamboo) [reply platforms/core-configuration/model-core/src/main/java/org/gradle/model/internal/asm/AsmClassGeneratorUtils.java:41]: Thanks, @abstratt!

**Comment by bamboo** [comment]
@bot-gradle merge

**Pull request merged**